DataFormats/L1TCalorimeterPhase2/interface/DigitizedClusterGT.h

```diff
@@ -0,0 +1,122 @@
+
+    // Constructor from digitized inputs
+    DigitizedClusterGT(ap_uint<1> isValid, ap_uint<16> pt, ap_uint<13> phi, ap_uint<14> eta, bool fullyDigitizedInputs) {
+      (void)fullyDigitizedInputs;
+      clusterData =
+          ((ap_uint<64>)isValid) | (((ap_uint<64>)pt) << 1) | (((ap_uint<64>)phi) << 17) | (((ap_uint<64>)eta) << 30);
+    }
+
+    // Constructor from float inputs that will perform digitization
+    DigitizedClusterGT(bool isValid, float pt_f, float phi_f, float eta_f) {
+      clusterData = (((ap_uint<64>)digitizeIsValid(isValid)) | ((ap_uint<64>)digitizePt(pt_f) << 1) |
+                     ((ap_uint<64>)digitizePhi(phi_f) << 17) | ((ap_uint<64>)digitizeEta(eta_f) << 30));
+    }
+
+    ap_uint<64> data() const { return clusterData; }
+
+    // Other getters
+    float ptLSB() const { return LSB_PT; }
+    float phiLSB() const { return LSB_PHI; }
+    float etaLSB() const { return LSB_ETA; }
+    ap_uint<1> isValid() const { return (clusterData & 0x1); }
+    ap_uint<16> pt() const { return ((clusterData >> 1) & 0xFFFF); }    // 16 1's = 0xFFFF
+    ap_uint<13> phi() const { return ((clusterData >> 17) & 0x1FFF); }  // (thirteen 1's)= 0x1FFF
+    ap_uint<12> phiMagnitude() const {
+      return ((clusterData >> 18) & 0xFFF);
+    }  // skip the LSB of the phi, and (twelve 1's)= 0xFFF
+    ap_uint<1> phiSign() const { return ((clusterData >> 17) & 0x1); }  // get the LSB of the phi
+    ap_uint<14> eta() const { return ((clusterData >> 30) & 0x3FFF); }  // (fourteen 1's) = 0x3FFF
+    ap_uint<13> etaMagnitude() const {
+      return ((clusterData >> 31) & 0x1FFF);
+    }  // skip the LSB of the eta, and (thirteen 1's) = 0x1FFF
+    ap_uint<1> etaSign() const { return ((clusterData >> 30) & 0x1); }  // get the LSB of the eta
+
+    float ptFloat() const { return (pt() * ptLSB()); }
+    float realPhi() const {  // convert from signed int to float
+      float sign = (phiSign() == 0) ? +1 : -1;
+      return (sign * phiMagnitude() * phiLSB());
+    }
+    float realEta() const {  // convert from signed int to float
+      float sign = (etaSign() == 0) ? +1 : -1;
+      return (sign * etaMagnitude() * etaLSB());
+    }
+    const int unusedBitsStart() const { return n_bits_unused_start; }  // unused bits start at bit 44
+
+    // Prints
+    void print(const std::string location);
+    void printIsValid(void) const { std::cout << "isValid: " << isValid() << std::endl; };
+    void printPt(void) const { std::cout << "pt: " << std::bitset<16>{pt()} << std::endl; };
+    void printPhi(void) const { std::cout << "phi: " << std::bitset<13>{phi()} << std::endl; };
+    void printEta(void) const { std::cout << "eta: " << std::bitset<14>{eta()} << std::endl; };
+    void printPtFloat(void) const { std::cout << "pt (GeV): " << ptFloat() << std::endl; };
+    void printRealPhi(void) const { std::cout << "real phi: " << realPhi() << std::endl; }
+    void printRealEta(void) const { std::cout << "real eta: " << realEta() << std::endl; }
```

Comment:
Same thing here

Reply:
Removed as of [e7e753d](https://github.com/cms-l1t-offline/cmssw/pull/1069/commits/e7e753df69e7aff845a3b3dfbaed9a06c9fdaec6)